### PR TITLE
Clustering step with unit tests complete

### PR DESCRIPTION
**Pull request by abenezer-gelaw**
In this PR, the clustering step is complete. There are unittests associated that check for appropriate assertions in edge cases. Additionally, a simple clustering is also done to make sure the DBSCAN call properly executes.

### CODE REVIEW

File: consistency/tool/clustering/clustering_test.json [hunk 1]
```diff
@@ -21,6 +21,25 @@
             "cleaned-words": []
          }
 
-    }
+    }, 
+    "cluster-words": {
```
[the remark this answers is not in the document]
Comment (abenezer-gelaw) [reply consistency/tool/clustering/clustering_test.json:25]: Added!

File: consistency/tool/clustering/clustering.py [hunk 2]
```diff
@@ -30,6 +30,30 @@ def clean_words(self):
...
+        data = np.arange(len(self.words)).reshape(-1, 1)
+    
+        try:
+            db = dbscan(data, metric=extract_indices_dice, eps=0.3, min_samples=2, algorithm='brute')
+        except Exception as e:
+            print(e, " Word Clustering Failed!")
+            return None
+    
+        #if all labels are -1, DBSCAN detected no possible clusters. 
+        if np.count_nonzero(db[1] == -1) == db[1].size:
```
Comment (shrutiparabgoogle) [inline consistency/tool/clustering/clustering.py:52]: Are we sure that dbscan always returns an array of at least length 1? If not, then `db[1]` will raise an exception here.

If it always returns a non-empty array, then lets's put a comment with a link to documentation which mentions that, else put a check on the length of `db`.

File: consistency/tool/clustering/clustering_test.json [hunk 3]
```diff
@@ -21,6 +21,25 @@
...
+      "warning-test1": {
```
Comment (shrutiparabgoogle) [inline consistency/tool/clustering/clustering_test.json:26]: NIT: Can we rename these as:

"warning-test1" -> "less-than-3-words"
"warning-test2" -> "null-words"

**Review by shrutiparabgoogle — COMMENTED** [review]
I have one blocking concern on the length of `db` object, otherwise looks good.

File: consistency/tool/clustering/clustering.py (hunk 2 above)
Comment (abenezer-gelaw) [reply consistency/tool/clustering/clustering.py:52]: As can be seen here (https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/cluster/_dbscan.py#:~:text=return%20est.core_sample_indices_%2C%20est.labels_) it always returns core indices and then labels. It appears there is no case for db[1] to raise an exception.
